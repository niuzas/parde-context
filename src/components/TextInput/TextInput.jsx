import React from 'react';
import PropTypes from 'prop-types';

const TextInput = ({ name, type, label, placeholder, handleChange }) => (
  <div className="field">
    {label && (
      <label className="label" htmlFor={name}>
        {label}
      </label>
    )}
    <div className="control">
      <input
        className="input"
        name={name}
        id={name}
        type={type}
        placeholder={placeholder}
        onChange={handleChange}
      />
    </div>
  </div>
);

const inputTypes = ['text', 'number', 'email', 'password'];

TextInput.propTypes = {
  name: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired,
  label: PropTypes.string,
  type: PropTypes.oneOf(inputTypes),
  handleChange: PropTypes.func
};

TextInput.defaultProps = {
  type: 'text',
  label: '',
  handleChange: () => null
};

export default TextInput;
