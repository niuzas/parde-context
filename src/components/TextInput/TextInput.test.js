import React from 'react';
import renderer from 'react-test-renderer';

import TextInput from './TextInput';

it('TextInput renders correctly', () => {
  const tree = renderer.create(<TextInput name="name" placeholder="placehodler" />).toJSON();
  expect(tree).toMatchSnapshot();
});
