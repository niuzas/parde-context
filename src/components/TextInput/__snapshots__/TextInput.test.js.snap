// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput renders correctly 1`] = `
<div
  className="field"
>
  
  <div
    className="control"
  >
    <input
      className="input"
      id="name"
      name="name"
      onChange={[Function]}
      placeholder="placehodler"
      type="text"
    />
  </div>
</div>
`;
